cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -334,11 +328,5 @@ def _has_requested_proctoring_provider_changed(current_provider, requested_provi
         """
         Return whether the requested proctoring provider is different than the current proctoring provider, indicating
         that the user has requested a change to the proctoring_provider Advanced Setting.
-        The requested_provider will be None if the proctoring_provider setting is not available (e.g. if the
-        ENABLE_PROCTORING_PROVIDER_OVERRIDES waffle flag is not enabled for the course). In this case, we consider
-        that there is no change in the requested proctoring provider.
         """
-        if requested_provider is None:
-            return False
-        else:
-            return current_provider != requested_provider
+        return current_provider != requested_provider
```

Comment:
Why does this function need to change? Is it now impossible that the provider would be falsey/`None` after we remove the provider overrides waffle?

I wonder whether what we're pulling out here also gives us backwards compatibility with old course data

Reply:
That's a good point, I will add those lines back in.